src/test/java/com/github/checkstyle/regression/git/DiffParserTest.java

```diff
@@ -20,47 +20,112 @@
+    public void testIsProperUtilsClass() throws Exception {
+        assertUtilsClassHasPrivateConstructor(DiffParser.class);
+    }
 
-    @Before
-    public void setUp() throws Exception {
-        repository = GitUtils.createNewRepository();
-        final File helloWorld = GitUtils.addAnEmptyFileAndCommit(repository, "HelloWorld");
-        GitUtils.createNewBranchAndCheckout(repository, "foo");
-        Files.write(helloWorld.toPath(), "hello world!".getBytes(), StandardOpenOption.APPEND);
-        GitUtils.addAllAndCommit(repository, "append text to HelloWorld");
+    @Test
+    public void testParseAddChange() throws Exception {
+        try (Repository repository = GitUtils.createNewRepository()) {
+            GitUtils.addAnEmptyFileAndCommit(repository, "HelloWorld");
+            GitUtils.createNewBranchAndCheckout(repository, "foo");
+            GitUtils.addAnEmptyFileAndCommit(repository, "AddedFile");
+            final List<GitChange> changes = DiffParser.parse(
+                    repository.getDirectory().getParent(), "foo");
+            assertEquals(1, changes.size());
+            assertChange(ADD, "AddedFile", changes.get(0));
+            FileUtils.deleteDirectory(repository.getDirectory().getParentFile());
+        }
     }
 
-    @After
-    public void tearDown() throws Exception {
-        repository.close();
+    @Test
+    public void testParseModifyChange() throws Exception {
+        try (Repository repository = GitUtils.createNewRepository()) {
+            final File helloWorld = GitUtils.addAnEmptyFileAndCommit(repository, "HelloWorld");
+            GitUtils.createNewBranchAndCheckout(repository, "foo");
+            Files.write(helloWorld.toPath(), "hello world!".getBytes(), StandardOpenOption.APPEND);
+            GitUtils.addAllAndCommit(repository, "append text to HelloWorld");
+            final List<GitChange> changes = DiffParser.parse(
+                    repository.getDirectory().getParent(), "foo");
+            assertEquals(1, changes.size());
+            assertChange(MODIFY, "HelloWorld", changes.get(0));
+            FileUtils.deleteDirectory(repository.getDirectory().getParentFile());
+        }
     }
 
     @Test
-    public void testIsProperUtilsClass() throws Exception {
-        assertUtilsClassHasPrivateConstructor(DiffParser.class);
+    public void testParseDeleteChange() throws Exception {
+        try (Repository repository = GitUtils.createNewRepository()) {
+            GitUtils.addAnEmptyFileAndCommit(repository, "HelloWorld");
+            GitUtils.addAnEmptyFileAndCommit(repository, "FileToDelete");
+            GitUtils.createNewBranchAndCheckout(repository, "foo");
+            GitUtils.removeFileAndCommit(repository, "FileToDelete");
+            final List<GitChange> changes = DiffParser.parse(
+                    repository.getDirectory().getParent(), "foo");
+            assertEquals(1, changes.size());
+            assertChange(DELETE, "/dev/null", changes.get(0));
```

Comment:
https://github.com/eclipse/jgit/blob/005e5feb4ecd08c4e4d141a38b9e7942accb3212/org.eclipse.jgit/src/org/eclipse/jgit/diff/DiffEntry.java#L373

In our previous discussion I want to keep "src/test" and deleted files mainly because I thought it is better to keep the possibility of getting all the changes, and leave a filter interface to let the real strategy code do decide which type of changes to use.

But I am re-thinking to drop deleted files now. If we would treat "ADD" and "MODIFY" the same way, then there is no need to distinguish the ChangeType. And "DELETE" is for sure useless as the file is gone.